indico/modules/users/controllers.py

```diff
@@ -177,13 +182,63 @@ class RHPersonalData(RHUserBase):
     def _process(self):
         form = UserDetailsForm(obj=FormDefaults(self.user, skip_attrs={'title'}, title=self.user._title),
                                synced_fields=self.user.synced_fields, synced_values=self.user.synced_values)
+        pic_form = UserPictureForm()
         if form.validate_on_submit():
             self.user.synced_fields = form.synced_fields
             form.populate_obj(self.user, skip=self.user.synced_fields)
             self.user.synchronize_data(refresh=True)
             flash(_('Your personal data was successfully updated.'), 'success')
             return redirect(url_for('.user_profile'))
-        return WPUser.render_template('personal_data.html', 'personal_data', user=self.user, form=form)
+        else:
+            if self.user.has_picture:
+                pic_form.picture.data = self.user
+        return WPUser.render_template('personal_data.html', 'personal_data', user=self.user, form=form,
+                                      pic_form=pic_form)
+
+
+class RHProfilePictureDisplay(RHUserBase):
+    def _process(self):
+        if not self.user.has_picture:
+            raise NotFound
+        metadata = self.user.picture_metadata
+        return send_file(metadata['filename'], BytesIO(self.user.picture), mimetype=metadata['content_type'],
```

Comment:
I don't think we use the original filename here. Imagine a user using a somewhat inappropriate name...

FWIW I think it isn't a good idea to send the original filename for the other images where we do this either (even though it was probably me writing some of those RHs 🙈) - storing them is one thing (and that's nice so the user sees their original filename), but for sending I'd use something generic, like "user.png"

Reply:
Agreed!